### PR TITLE
Ehance opt_as_ref_deref lint.

### LINKED ISSUE #5367: Enhance `option_as_ref_deref` lint with the |x| &**x case
In #4945, the `option_as_ref_deref` lint was added. This lint should be enhanced, so that it also detects:

```
opt.as_ref().map(|x| &**x)
```

_Originally posted by @Areredify in https://github.com/rust-lang/rust-clippy/pull/4945#issuecomment-577714718_

### PR DESCRIPTION
**Pull request by xiongmao86**
- [x] Added passing UI tests (including committed `.stderr` file)
- [x] `cargo test` passes locally
- [x] Run `cargo dev fmt`

Lint on opt.as_ref().map(|x| &**x). Fixes #5367.

changelog: lint on opt.as_ref().map(|x| &**x)

### CODE REVIEW

**Comment by xiongmao86** [comment]
What is the problem of Clippy dev test, I can't see details of why they fail. Just show `this check failed.`

**Comment by flip1995** [comment]
GHA is currently broken and we're waiting for a fix from the GH side.

**Comment by xiongmao86** [comment]
I will wait then.

File: tests/ui/option_as_ref_deref.stderr [hunk 1]
```diff
@@ -88,5 +88,17 @@ error: called `.as_mut().map(DerefMut::deref_mut)` (or with one of deref aliases
 LL |     let _ = opt.clone().as_mut().map(|x| x.deref_mut()).map(|x| x.len());
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using as_deref_mut instead: `opt.clone().as_deref_mut()`
 
-error: aborting due to 14 previous errors
+error: called `.as_ref().map(Deref::deref)` (or with one of deref aliases) on an Option value. This can be done more directly by calling `opt.as_deref()` instead
```
Comment (xiongmao86) [inline tests/ui/option_as_ref_deref.stderr:91]: @flip1995, Do I need to update `.as_ref().map(Deref::deref)` into `.as_ref().map(|x| &**x)` in this line?

Comment (flip1995) [reply tests/ui/option_as_ref_deref.stderr:91]: You can, if you want. It would be a nicer message. I'd merge this either way.

Comment (xiongmao86) [reply tests/ui/option_as_ref_deref.stderr:91]: Thanks, I'll try.

**Comment by xiongmao86** [comment]
Lint message refined.

**Comment by xiongmao86** [comment]
I am not very clear what to do on the failing test. Do I just wait until another rustup?

**Pull request merged**